### PR TITLE
Validate glue21

### PR DESCRIPTION
**Pull request by gwarf**
Validating GLUE 2.1 output with latest GLUE 2.1 draft.
Enabling some network-related information.

WIP, not ready for merge yet

### CODE REVIEW

**Comment by gwarf** [comment]
So it's probably now clean enough to be merged.
The default-network-type and public-network-name have to be provided in the conf file.
The network-in and network-out info wasn't tested yet as I'm not sure we have a VM in the FedCloud VO that includes those attributes.

**Comment by gwarf** [comment]
@enolfc can you please have a look? Do you think it's sufficient to merge those fixes for the time being?

File: cloud_info_provider/providers/openstack.py [hunk 1]
```diff
@@ -132,6 +132,8 @@ def get_compute_shares(self, **kwargs):
         shares = self.static.get_compute_shares(prefix=True)
         for share in shares.values():
             share['project'] = share.get('auth', {}).get('project_id')
+            share['default_network_type'] = share.get('default_network_type')
```
Comment (enolfc) [inline cloud_info_provider/providers/openstack.py]: what about OpenNebula? Shouldn't this be for both?

File: cloud_info_provider/providers/static.py [hunk 2]
```diff
@@ -131,7 +131,8 @@ def get_instances(self, **kwargs):
     def get_compute_shares(self, **kwargs):
         fields = ('instance_max_cpu', 'instance_max_ram',
                   'instance_max_accelerators',
-                  'auth', 'sla', 'network_info', 'membership')
+                  'auth', 'sla', 'network_info', 'default_network_type',
+                  'public_network_name', 'membership')
```
Comment (enolfc) [inline cloud_info_provider/providers/static.py]: I think this should be enough and the OpenStack code above is not needed

File: etc/templates/compute.glue21 [hunk 3]
```diff
@@ -364,10 +368,31 @@ GLUE2EntityOtherInfo: ${other}
 % endfor
 ## XXX Missing
 ## GLUE2CloudComputingImageContextualizationVersion:
+## XXX Not exposed for security reasons
 ## GLUE2CloudComputingImageDefaultUsername:
 ## GLUE2CloudComputingImageDefaultPassword:
-## GLUE2CloudComputingImageVersion:
 
+## All NetworkTraffic for $img_id
```
Comment (enolfc) [inline etc/templates/compute.glue21]: would be good to have some tests on the output of the templates, they keep growing and we do not have good understanding of the output

File: cloud_info_provider/providers/static.py (hunk 2 above)
Comment (gwarf) [reply cloud_info_provider/providers/static.py]: Yes, you are right, the openstack code was useless (was meant to be cutomized, but didn't customize anything). I've removed it.

File: cloud_info_provider/providers/openstack.py (hunk 1 above)
Comment (gwarf) [reply cloud_info_provider/providers/openstack.py]: Removed, static provider change is enough.

**Pull request merged**